backpack.tf extended sorting.user.js

```diff
@@ -1268,4 +1268,246 @@ class</a></li>
+                16 : 203,	// SMG
+                24 : 210,	// Revolver
+                4 : 194,	// Knife
+            }
+            this.nameiddict = {
+                "Scattergun" : 200,
+                "Force-A-Nature" : 45,
+                "Holy Mackerel" : 221,
+                "Bat" : 190,
+                "Rocket Launcher" : 205,
+                "Black Box" : 228,
+                "Shotgun" : 199,
+                "Flame Thrower" : 208,
+                "Backburner" : 40,
+                "Flare Gun" : 39,
+                "Axtinguisher" : 38,
+                "Grenade Launcher" : 206,
+                "Stickybomb Launcher" : 207,
+                "Chargin' Targe" : 131,
+                "Eyelander" : 132,
+                "Minigun" : 202,
+                "Gloves of Running Urgently" : 239,
+                "Frontier Justice" : 141,
+                "Crusader's Crossbow" : 305,
+                "Medi Gun" : 211,
+                "Bonesaw" : 198,
+                "Ubersaw" : 37,
+                "Sniper Rifle" : 201,
+                "Huntsman" : 56,
+                "SMG" : 203,
+                "Revolver" : 210,
+                "Ambassador" : 61,
+                "Knife" : 194,
+            }
+        }
+
+        getItemName(){
+            let itemName = this.name;
+            if(this.aussie == 1){
+                itemName = "Australium".concat(" ", itemName);
+            }
+            return itemName;
+        }
+
+        getItemId(){
+            let itemId;
+            let itemName = this.name.match(/((?<=((Blood)|(Gold)|(Silver)|(Rust)|(Carbonado)|(Diamond)) Botkiller ).+(?= Mk.II?))|((?<=Festive ).+)/g); // Regex courtesy of Moder112
+            if(itemName){
+                this.defindex = this.nameiddict[itemName[0]];
+                itemId = this.defindex;
+            }
+            else{
+                let defindexInt = parseInt(this.defindex);
+                if(this.idreplacedict.hasOwnProperty(defindexInt)){
+                    itemId = this.idreplacedict[defindexInt].toString();
+                }
+                else{
+                    itemId = this.defindex;
+                }
+            }
+            return itemId;
+        }
+
+        getItemSlot(){
+            let itemSlot;
+            switch(this.slot){
+                case "primary": case "secondary": case "melee": case "pda":
+                    itemSlot = 1;
+                    break;
+                case "misc":
+                    itemSlot = 0;
+                    break;
+                case "none":
+                    if(this.name == "Kit"){
+                        itemSlot = -1;
+                    }
+                    else if(this.name == "Fabricator"){
+                        itemSlot = -2;
+                    }
+                    else{
+                        itemSlot = 0;
+                    }
+                    break;
+            }
+            return itemSlot;
+        }
```

Comment:
(nitpick first)
As before, you can directly return the value of your switch case instead of storing it in a variable, then you also wouldn't need the break (nor the else condition)

---------------------------------------------

I think it would be clearer to return a string instead of a number which you later use in the switch statement, as this would be clearer for the reader of the code.

(For example "weapon", "misc", "kit", "fabricator" instead of 1,0,-1,-2)


Would look the following:

```suggestion
        getItemSlot(){
            switch(this.slot){
                case "primary": case "secondary": case "melee": case "pda":
                    return "weapon"
                case "misc":
                    return "misc"
                case "none":
                    if(this.name == "Kit")
                        return "kit"
                    if (this.name == "Fabricator") 
                        return "fabricator"
                    return "misc"
            }
        }
```